src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return getOptionalBook(resultSet,
+                        "Can't get data about book by id: " + id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't connect to DB "
+                    + "or SQL execute query is not correct", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> booksList = new ArrayList<>();
+        String sql = "SELECT * FROM books;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                booksList.add(getOptionalBook(resultSet,
+                        "Can't get data about book").orElseThrow());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get any books", e);
+        }
+        return booksList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int updatedRows = statement.executeUpdate();
+            if (updatedRows < 1) {
+                throw new DataProcessingException("Expected to update "
+                        + "at least to one row, but inserted 0 rows");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a book "
+                    + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            int updatedRows = statement.executeUpdate();
+            return updatedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book by id: " + id, e);
+        }
+    }
+
+    private Optional<Book> getOptionalBook(ResultSet resultSet, String textException) {
```

Comment:
just make this method throw an exception and catch it in findById()
```suggestion
    private Optional<Book> getOptionalBook(ResultSet resultSet, String textException) throws SQLException {
```